### PR TITLE
#21: Webm opus decoding

### PR DESCRIPTION
**Pull request by mstyura**
Integrated `libopus` to decode audio from `WebM` files with `OPUS` audio codec.

### CODE REVIEW

**Comment by mstyura** [comment]
@brion could you please review the changes?

**Comment by sushipan** [comment]
@mstyura 
thanks for your great joy !
OPUS audio codec works perfect on the demo, but when I play a vp9 codec 1920p or 4k video on the demo , the video frames are slower than the audio ,can you take a look on it ?

**Comment by brion** [comment]
@sushipan iirc 4k VPx is just too slow on most iOS devices. I can't get such resolutions to play reliably in VLC either, even on iPad pro... 1080p VP9 is likely too slow  for some devices too. However there's likely improvements that can be made, such as enabling multiple threads for VP9 files with tile columns. Can you open a fresh issue for the VP9 performance and I'll take a look after merging this?

**Comment by sushipan** [comment]
@brion
OK.

File: Classes/OGVDecoderWebM.m [hunk 1]
```diff
@@ -267,6 +289,44 @@ -(BOOL)processBegin
...
 #endif
+#ifdef OGVKIT_HAVE_OPUS_DECODER
+            if (audioCodec == NESTEGG_CODEC_OPUS) {
+                opusSampleRate = (opus_int32) audioParams.rate;
+                opusChannels = audioParams.channels;
+                int opusErr;
+                assert(opusDecoder == NULL);
+                opusDecoder = opus_decoder_create(opusSampleRate, opusChannels, &opusErr);
+                if (opusErr != OPUS_OK) {
+                    NSLog(@"Unable to initialize opus decoder with %@Hz and %@ channels. Error %d", opusSampleRate, opusChannels, opusErr);
```
Comment (brion) [inline Classes/OGVDecoderWebM.m]: "%@Hz and %@ channels" should be "%dHz and %d channels" I think as these are ints

**Pull request merged**